### DISCUSSION TITLE
javacpp: How can I NOT package the target native library?

### DISCUSSION THREAD

**Discussion by tlum** [bytedeco/javacpp-presets#1310] [Q&A] [answered]
I'm working on what is effectively a new [javacpp-presets](https://github.com/bytedeco/javacpp-presets) for [sane](https://gitlab.com/sane-project/backends), which is readily available from most Linux distro repositories. In addition to the core libsane.so back-end there are numerous manufacturer specific modules, and, some manufacturers publish their own proprietary modules as well, the packaging and distribution of which would violate their licensing.

Unfortunately, the proper way to prevent the packaging of libsane.so, which subsequently never manages to find the other modules, is not clear to me. Even when I don't build from source, and simply point to the system shared library, libsane.so gets copied to `sane/target/native/org/bytedeco/sane/linux-x86_64/libsane.so`, winds up in the jar from where it's loaded, and the back-end scanner modules are never found... which results in an API that supports no scanners at all. 

Right now I'm using this configuration in presets

    @Properties(
        value = {
            @Platform(
            		value = {"linux-x86_64"}
            		, includepath={"/usr/include/sane/","/usr/lib/x86_64-linux-gnu/sane"}
                    , linkpath = {"/usr/lib/x86_64-linux-gnu/sane"}
            		, include = {"<sane.h>","<saneopts.h>"}
            		, library = "libsane.so"
            		, link = "sane"
                    )
        },
        target = "org.bytedeco.sane",
        global = "org.bytedeco.sane.global.Sane"
    )

I should mention that I'm building with maven and presume that something additional is required in the pom.xml for a packaging goal at the very least, but it's also not clear what's required so the loader will find and load the sane library from it's native location, which in turn allows it to find and load it's optional children.

It's also desirable to detect a missing sane package and give a friendly message about making sure it's loaded in place of the less helpful UnsatisfiedLinkError messages someone would need to interpret, but first things first.

-TIA

**Comment by saudet** [comment]
JavaCPP will look for libraries in the normal java.library.path for JNI libraries, so the easiest thing to do is too make sure that java.library.path contains the libraries you wish to load.

If that doesn't work, please set the "org.bytedeco.javacpp.logger.debug" system property to "true" to get more information on the console.

**Comment by saudet** [comment, marked as answer]
To prevent copying libraries in the case of the Maven plugin, we can set copyLibs to false, for example:
https://github.com/bytedeco/javacpp-presets/blob/master/flycapture/pom.xml#L49

**Reply by saudet** [reply]
I think this is the answer you're looking for, but please let me know if this doesn't actually answer your question. Thanks!
